modules/soap-api/src/main/java/org/mule/runtime/soap/api/message/dispatcher/DefaultHttpMessageDispatcher.java

```diff
@@ -48,25 +57,40 @@ public DefaultHttpMessageDispatcher(HttpClient client) {
    */
   @Override
   public DispatchingResponse dispatch(DispatchingRequest context) {
+    InputStream content = retrieveLogging("Soap Request", context.getContent());
     MultiMap<String, String> parameters = new MultiMap<>();
     context.getHeaders().forEach(parameters::put);
     HttpRequest request = HttpRequest.builder()
         .uri(context.getAddress())
         .method(POST)
-        .entity(new InputStreamHttpEntity(context.getContent()))
+        .entity(new InputStreamHttpEntity(content))
         .headers(parameters)
         .build();
     try {
-      HttpResponse response = client.send(request, 5000, false, null);
-      InputStream content = response.getEntity().getContent();
-      return new DispatchingResponse(content, toHeadersMap(response));
+      HttpResponse response = client.send(request, DEFAULT_TIMEOUT_MILLIS, false, null);
+      return new DispatchingResponse(retrieveLogging("Soap Response", response.getEntity().getContent()), toHeadersMap(response));
     } catch (IOException e) {
       throw new DispatchingException("An error occurred while sending the SOAP request");
     } catch (TimeoutException e) {
       throw new DispatchingException("The SOAP request timed out", e);
     }
   }
 
+  /**
+   * Logs the content if it's log enabled.
+   */
+  private InputStream retrieveLogging(String title, InputStream content) {
+    if (LOGGER.isDebugEnabled()) {
+      String c = IOUtils.toString(content);
+      LOGGER.debug("Logging " + title);
```

Comment:
Logging doesn't mean anything. Add more context to identify the request. THink of a highly concurrent scenario, you'll need to trace each message to an event or request id

Reply:
makes sense, let me find a way to identify requests

Reply:
does the context headers have some kind of correlationId?